### PR TITLE
Added some comments to types and kernel

### PR DESCRIPTION
**Pull request by vgerber**
I used to bee an adventurer like u, but then the undocumented code drove me crazy

### CODE REVIEW

File: lib/isaac/isaac_types.hpp [hunk 1]
```diff
@@ -185,18 +190,23 @@ BOOST_PP_REPEAT( 3, ISAAC_DIM_DEF, ~ )
...
-    ISAAC_IDX_TYPE max_global_size;
-    isaac_size_dim < simdim > position;
-    isaac_size_dim < simdim > local_size;
-    isaac_size_dim < simdim > local_particle_size;
-    isaac_size_dim < simdim > global_size_scaled;
-    ISAAC_IDX_TYPE max_global_size_scaled;
-    isaac_size_dim < simdim > position_scaled;
-    isaac_size_dim < simdim > local_size_scaled;
+    isaac_size_dim < simdim > global_size;         //size of volume
+    ISAAC_IDX_TYPE max_global_size;                //? size if greatest dimension
```
Comment (ax3l) [inline lib/isaac/isaac_types.hpp]: ping @theZiz @FelixTUD: what is this?

Comment (theZiz) [reply lib/isaac/isaac_types.hpp]: Yes, this is the biggest of all dimensions of the volume above.

Comment (vgerber) [reply lib/isaac/isaac_types.hpp]: resolved in a1e1270fd313a036d071ff56174d7fc40f1272a9

**Comment by ax3l** [comment]
@theZiz seams you forgot to "approve" the PR :D

**Comment by theZiz** [comment]
> @theZiz seams you forgot to "approve" the PR :D

Nope, just haven't been in Germany for 2 weeks. :wink: 

@vgerber: Thanks for your work again! :smiley:

**Pull request merged**